src/app/space/create/deployments/apps/deployment-details.component.spec.ts

```diff
@@ -67,45 +72,33 @@ describe('DeploymentDetailsComponent', () => {
     mockSvc.deleteApplication.and.returnValue(Observable.of('mockDeletedMessage'));
 
     mockEnvironment = { name: 'mockEnvironment' } as Environment;
+  });
 
-    TestBed.configureTestingModule({
-      imports: [CollapseModule.forRoot()],
-      declarations: [
-        DeploymentDetailsComponent,
-        FakeDeploymentsDonutComponent,
-        FakeDeploymentGraphLabelComponent,
-        FakePfngChartSparkline
-      ],
-      providers: [
-        { provide: DeploymentsService, useValue: mockSvc }
-      ]
-    });
-
-    fixture = TestBed.createComponent(DeploymentDetailsComponent);
-    component = fixture.componentInstance;
-
+  initContext(DeploymentDetailsComponent, HostComponent, {
+    imports: [CollapseModule.forRoot()],
+    declarations: [
+      DeploymentDetailsComponent,
+      FakeDeploymentsDonutComponent,
+      FakeDeploymentGraphLabelComponent,
+      FakePfngChartSparkline
+    ],
+    providers: [
+      { provide: DeploymentsService, useFactory: () => mockSvc }
+    ]
+  }, component => {
     component.collapsed = false;
     component.applicationId = 'mockAppId';
     component.environment = mockEnvironment;
     component.spaceId = 'mockSpaceId';
-
-    fixture.detectChanges();
   });
 
   it('should generate unique chartIds for each DeploymentDetailsComponent instance', () => {
-    const DETAILS_COMPONENT_COUNT = 5;
-    const CHARTS_PER_DETAILS_COMPONENT = 2;
-    let uniqueIdSet = new Set();
-    times(DETAILS_COMPONENT_COUNT, () => {
-      let detailsComponent = new DeploymentDetailsComponent(mockSvc);
-      uniqueIdSet.add(detailsComponent.cpuConfig.chartId);
-      uniqueIdSet.add(detailsComponent.memConfig.chartId);
-    });
-    expect(uniqueIdSet.size).toBe(DETAILS_COMPONENT_COUNT * CHARTS_PER_DETAILS_COMPONENT);
+    let detailsComponent = new DeploymentDetailsComponent(mockSvc);
+    expect(detailsComponent.cpuConfig.chartId).not.toBe(detailsComponent.memConfig.chartId);
```

Comment:
Could you use the test Context here and check that the chartIds on the util-produced component are different, rather than instantiating a new one manually?